src/binary/create-logger.ts

```diff
@@ -1,16 +1,52 @@
 import winston from 'winston';
 
+import { LoggerFlags } from './types';
+import { resolveOption } from './utils/options/resolve-option';
+
 const levels = {
   error: 0,
   warn: 1,
   info: 2,
   verbose: 3,
   debug: 4,
 };
+type Level = keyof typeof levels;
+
+export const defaultLevel = 'info'; // if not provided
+
+function validateLevel(level: string | null): level is Level {
+  return level ? Object.keys(levels).includes(level) : false;
+}
+
+export function resolveLevel(
```

Comment:
It took me a few passes, but this does seem like solid logic and doing what was specified

Reply:
I can't read this either but it's well tested --> https://github.com/confio/ts-relayer/pull/106/files#diff-b3d42f480f31b66005903f1d5e9e38cebbf3df590a009b34edadd0acb793da81

Maybe we should simplify the spec to throw if multiple levels are provided, e.g.:
```bash
ibc-setup start --log-level verbose -qv # throw 
```

It would make the parsing easier.

Reply:
It looks good as it is. And the tests are top